xieffect/test/communities/tasks/question_test.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+from flask_fullstack import dict_cut, SocketIOTestClient
+
+from communities.tasks.tests_db import Question, QuestionKind
+from test.conftest import FlaskTestClient, delete_by_id
+
+
+@pytest.fixture()
+def question_data(test_id: int) -> dict[str, Any]:
+    return {
+        "text": "Test question",
+        "test_id": test_id,
+        "kind": QuestionKind.CHOICE.to_string(),
+    }
+
+
+@pytest.fixture()
+def question_id(question_data: dict[str, Any]) -> int:
+    question: Question = Question.create(
+        **dict(question_data, kind=QuestionKind.CHOICE)
+    )
+    yield question.id
+    delete_by_id(question.id, Question)
+
+
+@pytest.fixture()
+def question_ids(test_community: int, test_id: int, question_id: int) -> dict[str, int]:
+    return {
+        "community_id": test_community,
+        "test_id": test_id,
+        "question_id": question_id,
+    }
+
+
+def test_create_question(
+    client: FlaskTestClient,
+    socketio_client: SocketIOTestClient,
+    test_id: int,
+    test_community: int,
+    question_data: dict[str, Any],
+):
+    data: dict[str, Any] = {
+        "community_id": test_community,
+        **question_data,
+    }
+
+    socketio_client.assert_emit_ack(
+        event_name="new_question",
+        data=data,
+        expected_data={**dict_cut(data, "kind", "text"), "id": int},
+    )
+    client.get(
+        f"/communities/{test_community}/tests/{test_id}/",
+        expected_json={"questions": [dict_cut(data, "kind", "text")]},
+    )
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        pytest.param({"text": "update", "kind": "detailed"}, id="update_full_data"),
+        pytest.param({"kind": "simple"}, id="update_partial_data"),
+    ],
+)
+def test_update_question(
+    client: FlaskTestClient,
+    socketio_client: SocketIOTestClient,
+    test_id: int,
+    test_community: int,
+    question_ids: dict[str, int],
+    data: dict[str, str],
+):
+    socketio_client.assert_emit_ack(
+        event_name="update_question", data={**question_ids, **data}, expected_data=data
+    )
+
+    client.get(
+        f"/communities/{test_community}/tests/{test_id}/",
+        expected_json={"questions": [data]},
+    )
+
+
+def test_validation_kind(
+    socketio_client: SocketIOTestClient,
+    question_ids: dict[str, int],
+):
+    data: dict[str, Any] = {
+        **question_ids,
+        "text": "update",
+        "kind": "choice11223",
+    }
+    socketio_client.assert_emit_ack(
+        event_name="update_question",
+        data=data,
+        expected_code=400,
+        expected_message=f"{data['kind']} is not a valid {QuestionKind.__name__}",
+    )
```

Comment:
Давайте по максимуму всё в "одну строчку", тип так:
```suggestion
    socketio_client.assert_emit_ack(
        event_name="update_question",
        data={
            **question_ids,
            "text": "update",
            "kind": "choice11223",
        },
        expected_code=400,
        expected_message=f"{data['kind']} is not a valid QuestionKind",
    )
```

Reply:
Или не особо читается? Мне кажется так все на месте

Reply:
а {data['kind']} откуда возьмем если data удалить?